benchmarks/bench.hpp

```diff
@@ -72,3 +79,49 @@ std::tuple<uint64_t, uint64_t> bench_sort(const std::vector<T> arr,
             / lastfew;
     return std::make_tuple(avx_sort, std_sort);
 }
+template <typename K, typename V = uint64_t>
+std::tuple<uint64_t, uint64_t>
+bench_sort_kv(const std::vector<K> keys,
+              const std::vector<V> values,
+              const std::vector<sorted_t<K, V>> sortedaar,
+              const uint64_t iters,
+              const uint64_t lastfew)
+{
+
+    std::vector<K> keys_bckup = keys;
+    std::vector<V> values_bckup = values;
+    std::vector<sorted_t<K, V>> sortedaar_bckup = sortedaar;
+
+    std::vector<uint64_t> runtimes1, runtimes2;
+    uint64_t start(0), end(0);
+    for (uint64_t ii = 0; ii < iters; ++ii) {
+        start = cycles_start();
+        avx512_qsort_kv<K>(
+                keys_bckup.data(), values_bckup.data(), keys_bckup.size());
+        end = cycles_end();
+        runtimes1.emplace_back(end - start);
+        keys_bckup = keys;
+        values_bckup = values;
+    }
+    uint64_t avx_sort = std::accumulate(runtimes1.end() - lastfew,
+                                        runtimes1.end(),
+                                        (uint64_t)0)
+            / lastfew;
+
+    for (uint64_t ii = 0; ii < iters; ++ii) {
+        start = cycles_start();
+        std::sort(sortedaar_bckup.begin(),
```

Comment:
Not sure if std::sort is a fair comparison. The way the key-values are arranged in memory for `std::vector<sorted_t<K, V>>` is very different when compared two distinct vectors  `std::vector<K>, std::vector<V>`. I would think this can alter performance significantly because you can no longer load continuous chunk of memory into ZMM registers. Does Oceanbase have two separate vectors for key and values (I assume they do cos that's the way you implemented the sort). How does Oceanbase currently sort them?

Reply:
ObStoreRow is the base unit to do the sort in Oceanbase. The key and value are all ObObject class type in the ObStoreRow. There is no sperate vector for key and value. The ObStoreRow is passed to the std::sort in Oceanbase. std::sort first get the key type from ObObject and then do the sort work.

Reply:
To enable the AVX512 sort, we need extract the key from the ObStoreRow and then pass the key and ObStoreRow as the value to avx512_sort(key, value) like API. That's way we need the key-value interface